### PR TITLE
[WIP] ePub generation fixes

### LINKED ISSUE #5707: Epub: TemplateAssertionError: no test named 'index' (2.8)
I get error below while trying to generate epub file:

```
# Sphinx version: 1.3.1
# Python version: 2.7.10 (CPython)
# Docutils version: 0.12 release
# Jinja2 version: 2.7.3
# Last messages:
#
#   looking for now-outdated files...
#   none found
#   pickling environment...
#   done
#   checking consistency...
#   done
#   preparing documents...
#   done
#   writing output... [  0%] best_practices/business-logic
# Loaded extensions:
#   sensio.sphinx.phpcode (unknown version) from /Library/Python/2.7/site-packages/sensio/sphinx/phpcode.pyc
#   sphinx.ext.autodoc (1.3.1) from /Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/ext/autodoc.pyc
#   sensio.sphinx.configurationblock (unknown version) from /Library/Python/2.7/site-packages/sensio/sphinx/configurationblock.pyc
#   sphinx.ext.doctest (1.3.1) from /Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/ext/doctest.pyc
#   alabaster (0.7.6) from /Library/Python/2.7/site-packages/alabaster-0.7.6-py2.7.egg/alabaster/__init__.pyc
#   sensio.sphinx.bestpractice (unknown version) from /Library/Python/2.7/site-packages/sensio/sphinx/bestpractice.pyc
#   sensio.sphinx.refinclude (unknown version) from /Library/Python/2.7/site-packages/sensio/sphinx/refinclude.pyc
#   symfonycom.sphinx (unknown version) from /Users/jacek/Downloads/symfony-docs/_theme/_exts/symfonycom/sphinx/__init__.pyc
#   sphinx.ext.todo (1.3.1) from /Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/ext/todo.pyc
#   sensio.sphinx.codeblock (unknown version) from /Library/Python/2.7/site-packages/sensio/sphinx/codeblock.pyc
Traceback (most recent call last):
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/cmdline.py", line 245, in main
    app.build(opts.force_all, filenames)
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/application.py", line 264, in build
    self.builder.build_update()
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/__init__.py", line 245, in build_update
    'out of date' % len(to_build))
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/__init__.py", line 316, in build
    self.write(docnames, list(updated_docnames), method)
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/__init__.py", line 354, in write
    self._write_serial(sorted(docnames), warnings)
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/__init__.py", line 362, in _write_serial
    self.write_doc(docname, doctree)
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/epub.py", line 378, in write_doc
    return StandaloneHTMLBuilder.write_doc(self, docname, doctree)
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/html.py", line 445, in write_doc
    self.handle_page(docname, ctx, event_arg=doctree)
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/epub.py", line 465, in handle_page
    outfilename, event_arg)
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/builders/html.py", line 786, in handle_page
    output = self.templates.render(templatename, ctx)
  File "/Library/Python/2.7/site-packages/Sphinx-1.3.1-py2.7.egg/sphinx/jinja2glue.py", line 138, in render
    return self.environment.get_template(template).render(context)
  File "/Library/Python/2.7/site-packages/jinja2/environment.py", line 969, in render
    return self.environment.handle_exception(exc_info, True)
  File "/Library/Python/2.7/site-packages/jinja2/environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/jacek/Downloads/symfony-docs/_theme/_templates/layout.html", line 6, in <module>
    {% set isIndex = pagename is index %}
TemplateAssertionError: no test named 'index'
```

Any hints regarding that?

### PR DESCRIPTION
**Pull request by michaelperrin**
This PR enables back the generation of ePub files. It fixes #5707 and #6485.

The main problems were due to:

* A missing reference for templates in the _symfonycom_ extension.
* Several syntax errors in the HTML files (missing closing tags, lack of quotes on attribute values).

Note that I didn't change the strange need of a double `<body>` tag (yup, that's really strange, but there must be a reason).

Generating the ePub file can be done with the following command:

    make epub

I have tested the generated file in Apple iBooks, and it works fine.

I also removed the changelog from the generated file, as it was making a really big file without any added value. I can put it back if you wish.
_EDIT: I put it back for now_

I can also squash the commits into one.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -3,11 +3,6 @@
 Symfony Documentation
 =====================
 
-.. toctree::
-    :hidden:
-
-    changelog
```
Comment (xabbuh) [inline index.rst]: Looks like you have to keep this (see the failing build on Travis CI).

**Review by xabbuh — CHANGES_REQUESTED** [review]

File: index.rst (hunk 1 above)
Comment (michaelperrin) [inline index.rst]: This was intentional as the generated file contains the whole changelog, which is huge and not really useful. But I'm going to remove it for now, and we can decide later if it is relevant to keep it.

**Comment by michaelperrin** [comment]
@xabbuh I removed the changelog removal (!) for now.

**Comment by sepehr** [comment]
Would love to see this merged. Big thanks, @michaelperrin.

**Comment by michaelperrin** [comment]
Thanks @sepehr . The generated ePub file could be better, but this PR can't do any harm anyway as it basically fixes HTML issues is the docs (missing closing tags, missing quotes for HTML attributes, etc.).

**Comment by sepehr** [comment]
Yeah, even with these fixes, there are a lot of problems with the generated epub. `epubcheck` throws a lot of errors. And the Google Play Books app cannot parse the epub even though it seems to be working with Apple's iBook.

@michaelperrin, Is there any chance that you have working compiled epub of Symfony 3.x docs? If so, I'd be grateful if you share it.

**Comment by michaelperrin** [comment]
@sepehr I didn't know the `epubcheck` tool and there are indeed a huge amount of errors in the generated file! I'll have a look at it but I think it needs a lot of work to make all the changes...

**Comment by javiereguiluz** [comment]
@michaelperrin I'm sorry we didn't merge this on time. If you could rebase it, I'd merge it quickly. I guess there will be still some issues in the generated ePub, but as you said, there are too many to fix them all. the work you did here is good enough to fix some things. Thank you.

**Comment by javiereguiluz** [comment]
Let's close this one because we have problems even to generate HTML and PDF ... so we can't dedicate resources to a new format. I'm sorry.

**Comment by michaelperrin** [comment]
@javiereguiluz Yep, better closing it indeed.
If needed, I can make the changes in an other PR to fix the few issues as you requested.

**Pull request closed without merging**